src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -21,60 +22,230 @@
+                            }
+                            else {
+                                updateTrackingInfo(statusType, oldType, statusDescription, trackingId, date, SQL_UPDATE_UPS_RETURN_STATUS, isReturn, kit);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    private static void updateTrackingInfo(String statusType, String oldType, String statusDescription, String trackingId, String date, String query, boolean isReturn, DdpKit kit) {
+        String upsUpdate = statusType + " " + statusDescription;
+        SimpleResult result = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, upsUpdate);
+                stmt.setString(2, date);
+                stmt.setString(3, trackingId);
+                int r = stmt.executeUpdate();
+                if (r != 2) {//number of subkits
+                    throw new RuntimeException("Update query for UPS tracking updated " + r + " rows!");
+                }
+            }
+            catch (Exception e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (result.resultException != null) {
+            throw new RuntimeException(result.resultException);
+        }
+        else {
+            if (!isReturn) {
+                //if delivered notif pepper
+                if (DELIVERY.equals(statusType) && !(DELIVERY.equals(oldType))) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for delivered kit with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(kitDDPNotification);
+
+                    }
+                    else {
+                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
+                    }
+                }
 
             }
-            catch (IOException e) {
-                throw new RuntimeException("couldn't get response from ups tracking ", e);
+            else {
+                //if picked up place order
+                if (!(PICKUP.equals(oldType)) && !(IN_TRANSIT.equals(oldType)) && (PICKUP.equals(statusType) || IN_TRANSIT.equals(statusType))) {
+                    Instant now = Instant.now();
+                    orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getKitLabel(), kit.getExternalOrderNumber(), now);
```

Comment:
Is `kit.getKitLabel()` the `TBOS-` value?  From what I saw earlier in the database, this was a different label.  The `TBOS-` label is what's required here, since that ends up in `ORC-2`.

Reply:
Yes I changed that

Reply:
We should probably do some sort of duplicate check here--can we keep track of which kits we've ordered via `orderTest()` and only place an order for kits that don't already have an order in CE?  Placing multiple orders for the same thing in CE is to be avoided, and although Birch does some duplicate filtering of its own, we should put duplicate detection here as well.

Reply:
Maybe we should store `OrderResponse.handle` in DSM, and if we see a non-blank `handle`, we don't place an order?

Reply:
I can keep track in dsm, would be easier to check. Does CE gives us an id o keep or I can just have a flag for the time order placed

Reply:
`OrderResponse.handle` will be different even for a duplicated order, but just knowing that there's _some_ id should be sufficient.